components/apimgt/org.wso2.carbon.apimgt.rest.api/src/main/java/org/wso2/carbon/apimgt/rest/api/handlers/BasicAuthenticationHandler.java

```diff
@@ -0,0 +1,126 @@
+
+        if (StringUtils.isEmpty(username)) {
+            logger.error("username is null/empty.");
+            return Response.status(Response.Status.UNAUTHORIZED).type(MediaType.APPLICATION_JSON).entity("Username " +
+                    "cannot be null").build();
+        } else if (StringUtils.isEmpty(password)  && certObject == null ) {
+            logger.error("password is null/empty.");
+            return Response.status(Response.Status.UNAUTHORIZED).type(MediaType.APPLICATION_JSON).entity("password " +
+                    "cannot be null").build();
+        }
+
+        return Authenticate(certObject, username, password);
+    }
+
+    /**
+     * Authenticate with the user credentials.
+     *
+     * @param certObject   Certificate object of the request
+     * @param username     Username
+     * @param password     Password
+     * @return             Response, if unauthorized. Null, if Authorized.
+     */
+    private Response Authenticate(Object certObject, String username, String password){
+        PrivilegedCarbonContext carbonContext = PrivilegedCarbonContext.getThreadLocalCarbonContext();
+        RealmService realmService = (RealmService)carbonContext.getOSGiService(RealmService.class, null);
+        RegistryService registryService = (RegistryService) carbonContext.getOSGiService(RegistryService.class, null);
+        String tenantDomain = MultitenantUtils.getTenantDomain(username);
+        int tenantId;
+        try {
+            tenantId = realmService.getTenantManager().getTenantId(tenantDomain);
+            UserRealm userRealm = null;
+            if (certObject == null) {
+                userRealm = AnonymousSessionUtil.getRealmByTenantDomain(registryService, realmService, tenantDomain);
+                if (userRealm == null) {
+                    logger.error("Invalid domain or unactivated tenant login");
+                    return Response.status(Response.Status.UNAUTHORIZED).type(MediaType.APPLICATION_JSON).entity(
+                            "Tenant not found").build();
+                }
+            }
+            username = MultitenantUtils.getTenantAwareUsername(username);
+            // if authenticated
+            if (certObject != null || userRealm.getUserStoreManager().authenticate(username, password)) {
+                // set the correct tenant info for downstream code.
+                carbonContext.setTenantDomain(tenantDomain);
+                carbonContext.setTenantId(tenantId);
+                carbonContext.setUsername(username);
+                return null;
+            } else {
+                logger.error(String.format("Authentication failed. Please check your username/password"));
+                return Response.status(Response.Status.UNAUTHORIZED).type(MediaType.APPLICATION_JSON).entity(
+                        "Authentication failed. Please check your username/password").build();
+            }
+        } catch (Exception e) {
```

Comment:
Fixed in #1528 (https://github.com/wso2/carbon-apimgt/commit/6304aaf743e6869aaa872e56f908651fe58927cd)